### PR TITLE
tools: Fix SELinux policy update hack in %post

### LINKED ISSUE #13183: cockpit-tls: connect(/run/cockpit/wsinstance/https-factory.sock) failed
Cockpit version: 207
OS: Fedora 31

Throws an error on server if accessing cockpit with https://ip:9090 from browser.

```
systemd[1]: Starting Cockpit Web Service...
systemd[1]: Started Cockpit Web Service.
cockpit-tls[1117]: cockpit-tls: gnutls_handshake failed: A TLS fatal alert has been received.
cockpit-tls[1117]: cockpit-tls: gnutls_handshake failed: A TLS fatal alert has been received.
cockpit-tls[1117]: cockpit-tls: connect(/run/cockpit/wsinstance/https-factory.sock) failed: Permission denied
```

```
> ls -laZ /run/cockpit/wsinstance/https-factory.sock
srw-------. 1 cockpit-ws cockpit-ws system_u:object_r:cockpit_var_run_t:s0 0 /run/cockpit/wsinstance/https-factory.sock

>ps aux | grep -i cockpit
cockpit-ws              1440  0.0  0.0   8512  4096 ?        Ss   cockpit-tls
```

It runs when I disable SELINUX:
```
/etc/sysconfig/selinux
   SELINUX=permissive
```

```
>cat /var/log/audit/audit.log | grep -i /https-factory.sock
type=AVC msg=audit: avc:  denied  { connectto } for  pid=1080 comm="cockpit-tls" path="/run/cockpit/wsinstance/https-factory.sock" scontext=system_u:system_r:cockpit_ws_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=unix_stream_socket permissive=1
```
  
I've upgraded from Fedora 29 to Fedora 31.

Does anyone know how to fix it? Thanks.

### PR DESCRIPTION
**Pull request by martinpitt**
Applying this hack really isn't optional, cockpit does not work without
it if SELinux is active. Thus remove the condition, and add the
necessary package dependencies to ensure that restorecon, semanage, and
friends are available.

Only add the new dependencies on release builds though, not on CI
builds. Our test images already have the necessary packages (otherwise
fedora-{30,31} tests would completely fail), but our images' cockpit/ws
container instances don't -- and we also don't need the %post in a
container. This hack can be dropped again once a recent selinux-policy
lands in Fedora 30, i. e. in a few weeks at most.

Drop the "fedora == 29" check, as we haven't released to Fedora 29 for a
while already, and it is EOL now.

Fixes #13183
https://bugzilla.redhat.com/show_bug.cgi?id=1777001

### CODE REVIEW

File: tools/cockpit.spec [hunk 1]
```diff
@@ -450,15 +455,14 @@ getent passwd cockpit-wsinstance >/dev/null || useradd -r -g cockpit-wsinstance
 # firewalld only partially picks up changes to its services files without this
 test -f %{_bindir}/firewall-cmd && firewall-cmd --reload --quiet || true
 
-%if 0%{?rhel} || 0%{?fedora} == 29
+%if 0%{?rhel}
```
Comment (martinpitt) [inline tools/cockpit.spec]: Forgot to mention that in the commit message (happy to do on squashing): F29 is EOL now, and we stopped uploading to F29 some months ago already.

**Comment by martinpitt** [comment]
Argh, this [does not work](https://209.132.184.41:8493/logs/pull-13196-20191127-074444-2362dfda-cockpit-project-cockpit--fedora-coreos/log.html) on fedora-coreos, and we also don't need it in the container.

**Comment by martinpitt** [comment]
I started looking into sprinkling `--no-deps` into our test scripts for fixing coreos and bastion, but that's going to become too ugly and intrusive. This hack only needs to last for a few weeks, so let's go with only applying the dependencies for release builds.

**Comment by mvollmer** [comment]
Tests are failing: 
```
error: /var/tmp/srpm-build.nQsEdF/cockpit.spec:403: bad %if condition
```

**Comment by martinpitt** [comment]
@mvollmer: Argh @ clumsy rpm macro syntax. Fixed now, PTAL?

**Pull request merged**